datadelivery/s3.py

```diff
@@ -40,7 +40,18 @@ def _check_response(response):
         try:
             response.raise_for_status()
         except requests.HTTPError:
-            raise S3Exception(response.text)
+            raise S3Exception(S3.make_message_for_http_error(response))
+
+    @staticmethod
+    def make_message_for_http_error(response):
+        message = response.text
+        try:
+            data = response.json()
+            if 'detail' in data:
+                message = data['detail']
+        except ValueError:
+            pass  # response was not JSON
+        return message
 
     def _get_current_endpoint(self):
```

Comment:
This note is about line 64, still error-handling

I updated my local `.datadelivery.yml` to have a token for the live server. When I try to run a delivery, I get the following error:

```
$ datadelivery deliver -b sample --email first.last@duke.edu
Traceback (most recent call last):
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/bin/datadelivery", line 11, in <module>
    load_entry_point('datadelivery==0.0.1', 'console_scripts', 'datadelivery')()
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/__main__.py", line 14, in main
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/argparser.py", line 26, in parse_and_run_commands
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/argparser.py", line 73, in _run_deliver
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/commands.py", line 24, in deliver
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/commands.py", line 14, in _create_s3
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/s3.py", line 11, in __init__
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/s3.py", line 64, in _get_current_endpoint
AttributeError: 'Config' object has no attribute 's3_url'
```

Reply:
I made a change to fix this bug.